lib/improver/wind_direction.py

```diff
@@ -345,71 +407,50 @@ def process(cube_ens_wdir):
         # Force input cube to float32.
         enforce_float32_precision(cube_ens_wdir)
 
-        # Creates cubelists to hold data.
-        wdir_cube_list = iris.cube.CubeList()
-        r_vals_cube_list = iris.cube.CubeList()
-        confidence_measure_cube_list = iris.cube.CubeList()
-
         y_coord_name = cube_ens_wdir.coord(axis="y").name()
         x_coord_name = cube_ens_wdir.coord(axis="x").name()
-        for slice_ens_wdir in cube_ens_wdir.slices(["realization",
-                                                    y_coord_name,
-                                                    x_coord_name]):
+        for wdir_slice in cube_ens_wdir.slices(["realization",
+                                                y_coord_name,
+                                                x_coord_name]):
+            self._reset()
+            self.wdir_slice = wdir_slice
```

Comment:
I don't think this needs to be a class member - and I think it makes the code slightly clearer if it isn't (as it doesn't have to be covered in "reset", for example).

As a general comment, I'm finding it slightly difficult to interpret what "process" is doing, partly because there are a lot of class members and it's hard to extract the key focus points.  So I wonder if there's any variable or function renaming that might clarify the process, and whether we can be any more selective about the class members.

Reply:
Agreed.